File: app/src/main/java/me/zhanghai/android/files/navigation/NavigationRoot.java
/*
 * Copyright (c) 2018 Zhang Hai <Dreaming.in.Code.ZH@Gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.navigation;

import android.content.Context;

import androidx.annotation.NonNull;
import java8.nio.file.Path;

public interface NavigationRoot {

    @NonNull
    Path getPath();

    @NonNull
    String getName(@NonNull Context context);
}

File: app/src/main/java/me/zhanghai/android/files/navigation/NavigationRoot.kt
/*
 * Copyright (c) 2018 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.navigation

import android.content.Context
import java8.nio.file.Path

interface NavigationRoot {
    val path: Path

    fun getName(context: Context): String
}
